query/QueryManager.java

```diff
@@ -48,78 +44,60 @@ public QueryManager(final RPCTransaction rpcTransaction) {
         this.rpcTransaction = rpcTransaction;
     }
 
-    public QueryFuture<Stream<ConceptMap>> match(final GraqlMatch query) {
+    public Stream<ConceptMap> match(final GraqlMatch query) {
```

Comment:
We change `QueryFuture<Stream>` to just `Stream`, because `Stream` provides us the behaviour of "send a request and return answers only when you specifically ask for them (by iterating)" that we want.